### PR TITLE
feat: Avatar.Group support shape props

### PR DESCRIPTION
**Pull request by li-jia-nan**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 💡 Background and solution

- #43082

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | feat: Avatar.Group support shape props |
| 🇨🇳 Chinese | Avatar.Group 支持设置形状 |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 8060644</samp>

Added a `shape` prop to the `Avatar` and `Avatar.Group` components to enable square avatars. Updated the `Avatar` component to use the new `AvatarContext` from the `SizeContext` file. Added a new demo and documentation for the `shape` prop. Fixed some minor style issues.

### <samp>🤖 Generated by Copilot at 8060644</samp>

*  Add `shape` prop to `Avatar` and `Avatar.Group` components to allow square avatars ([link](https://github.com/ant-design/ant-design/pull/43817/files?diff=unified&w=0#diff-a9110968bd15226a58d0270285b7d34c0c37c415a6e5ca231c84fb3fb82ef9efL10-R10), [link](https://github.com/ant-design/ant-design/pull/43817/files?diff=unified&w=0#diff-a9110968bd15226a58d0270285b7d34c0c37c415a6e5ca231c84fb3fb82ef9efL47-R47), [link](https://github.com/ant-design/ant-design/pull/43817/files?diff=unified&w=0#diff-a9110968bd15226a58d0270285b7d34c0c37c415a6e5ca231c84fb3fb82ef9efL109-R109), [link](https://github.com/ant-design/ant-design/pull/43817/files?diff=unified&w=0#diff-a9110968bd15226a58d0270285b7d34c0c37c415a6e5ca231c84fb3fb82ef9efL149-R157), [link](https://github.com/ant-design/ant-design/pull/43817/files?diff=unified&w=0#diff-fc956249327d751be0b3a034d627120925a45b9c6bab42bdf56d3c9c8074f454R27), [link](https://github.com/ant-design/ant-design/pull/43817/files?diff=unified&w=0#diff-fc956249327d751be0b3a034d627120925a45b9c6bab42bdf56d3c9c8074f454L35-R43), [link](https://github.com/ant-design/ant-design/pull/43817/files?diff=unified&w=0#diff-fc956249327d751be0b3a034d627120925a45b9c6bab42bdf56d3c9c8074f454L77-R81), [link](https://github.com/ant-design/ant-design/pull/43817/files?diff=unified&w=0#diff-fc956249327d751be0b3a034d627120925a45b9c6bab42bdf56d3c9c8074f454L86-R90), [link](https://github.com/ant-design/ant-design/pull/43817/files?diff=unified&w=0#diff-fd35a1f57adc1678e98773049f2bb7dbf8b74ff319e1734b9d4e3413ac3efb5bR56), [link](https://github.com/ant-design/ant-design/pull/43817/files?diff=unified&w=0#diff-0a02c7f9917c7619e1ecea4387136e7129ebede165daa9fb2ee90f66f54fbb2bR61))
*  Use `AvatarContext` type to pass `shape` and `size` props from `Avatar.Group` to `Avatar` ([link](https://github.com/ant-design/ant-design/pull/43817/files?diff=unified&w=0#diff-cf0660cbe23845d8a3aecc65fdab53fc209075ca94b0136b410aab968986e147L6-R25), [link](https://github.com/ant-design/ant-design/pull/43817/files?diff=unified&w=0#diff-a9110968bd15226a58d0270285b7d34c0c37c415a6e5ca231c84fb3fb82ef9efL10-R10), [link](https://github.com/ant-design/ant-design/pull/43817/files?diff=unified&w=0#diff-fc956249327d751be0b3a034d627120925a45b9c6bab42bdf56d3c9c8074f454L77-R81), [link](https://github.com/ant-design/ant-design/pull/43817/files?diff=unified&w=0#diff-fc956249327d751be0b3a034d627120925a45b9c6bab42bdf56d3c9c8074f454L86-R90))
*  Simplify `cloneElement` function in `Avatar.Group` to only pass `key` prop to children ([link](https://github.com/ant-design/ant-design/pull/43817/files?diff=unified&w=0#diff-fc956249327d751be0b3a034d627120925a45b9c6bab42bdf56d3c9c8074f454L54-R61))
*  Add default values for `maxPopoverPlacement` and `maxPopoverTrigger` props in `Avatar.Group` ([link](https://github.com/ant-design/ant-design/pull/43817/files?diff=unified&w=0#diff-fc956249327d751be0b3a034d627120925a45b9c6bab42bdf56d3c9c8074f454L35-R43))
*  Use destructured `style` prop in `Avatar.Group` ([link](https://github.com/ant-design/ant-design/pull/43817/files?diff=unified&w=0#diff-fc956249327d751be0b3a034d627120925a45b9c6bab42bdf56d3c9c8074f454L35-R43), [link](https://github.com/ant-design/ant-design/pull/43817/files?diff=unified&w=0#diff-fc956249327d751be0b3a034d627120925a45b9c6bab42bdf56d3c9c8074f454L77-R81), [link](https://github.com/ant-design/ant-design/pull/43817/files?diff=unified&w=0#diff-fc956249327d751be0b3a034d627120925a45b9c6bab42bdf56d3c9c8074f454L86-R90))
*  Add demo for `shape` prop in `Avatar.Group` in `components/avatar/demo/group.tsx` ([link](https://github.com/ant-design/ant-design/pull/43817/files?diff=unified&w=0#diff-72a8aca6110a0e25998aabdd9f65e6d593252db59edbef23c231d2b4987c26ffR53-R59))

### CODE REVIEW

File: components/avatar/group.tsx [hunk 1]
```diff
@@ -74,17 +77,17 @@ const Group: React.FC<GroupProps> = (props) => {
       </Popover>,
     );
     return wrapSSR(
-      <SizeContextProvider size={size}>
-        <div className={cls} style={props.style}>
+      <SizeContextProvider size={size} shape={shape}>
```
Comment (MadCcc) [inline components/avatar/group.tsx:80]: 这个 Context 得换个名字

Comment (li-jia-nan) [reply components/avatar/group.tsx]: done

**Review by kiner-tang — APPROVED** [review]

File: components/avatar/SizeContext.tsx [hunk 2]
```diff
@@ -3,16 +3,25 @@ import type { ScreenSizeMap } from '../_util/responsiveObserver';
...
+export const AvatarContextProvider: React.FC<SizeContextProps> = ({ children, size, shape }) => (
+  <AvatarContext.Provider
+    value={React.useMemo<AvatarContextType>(() => ({ size, shape }), [size, shape])}
+  >
+    {children}
+  </AvatarContext.Provider>
+);
 
-export default SizeContext;
+export default AvatarContext;
```
Comment (MadCcc) [inline components/avatar/SizeContext.tsx:27]: 文件名也改了吧

Comment (li-jia-nan) [reply components/avatar/SizeContext.tsx]: done

File: components/avatar/avatarContext.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+export const AvatarContextProvider: React.FC<SizeContextProps> = ({ children, size, shape }) => (
+  <AvatarContext.Provider
+    value={React.useMemo<AvatarContextType>(() => ({ size, shape }), [size, shape])}
+  >
+    {children}
+  </AvatarContext.Provider>
+);
+
+export default AvatarContext;
```
Comment (MadCcc) [inline components/avatar/avatarContext.tsx:27]: AvatarContext 或者 context，命名统一一下

Comment (li-jia-nan) [reply components/avatar/avatarContext.tsx]: 把 `AvatarContextProvider` 删掉了，直接在消费的地方用就行，没必要封装一层

**Review by MadCcc — APPROVED** [review]

**Pull request merged**